scripts/ensure-bundle-deps.ts

```diff
@@ -0,0 +1,146 @@
+/* eslint-disable no-console */
+import * as childProcess from 'child_process';
+import * as fs from 'fs';
+import * as path from 'path';
+
+type ProcessDotEnv = typeof process.env & {
+  INIT_CWD: string;
+  npm_config_argv: string;
+};
+
+/**
+ * Ensure that `build:bundle` has all of the dependencies it needs to run. Works at both the repo and package level.
+ */
+export async function ensureBundleBuildPrereqs(options: {
+  dependencies: string[];
+  maxRetries?: number;
+}): Promise<void> {
+  const { maxRetries = 12, dependencies } = options;
+
+  const processDotEnv = process.env as ProcessDotEnv;
```

Comment:
I mean I would rather just treat these as possible to be undefined and do existence checks on `INIT_CWD` and `npm_config_argv` instead of using a custom `ProcessDotEnv` type. Let's just play on the safe side here.

```ts
const initCwd = process.env.INIT_CWD;
if (initCwd === undefined) {
  console.error("INIT_CWD environment variable not defined");
  return;
}

const npmConfigArgv = process.env.npm_config_argv;
if (npmConfigArgv === undefined) {
  console.error("npm_config_argv environment variable not defined");
  return;
}
```

Reply:
IDK, I feel like the only way this breaks is if we upgrade yarn (at which point we'd know to be watching for breakage), and the worst that happens if it does is our build won't go. Feels fairly low stakes to me. But, anyway, added a check.